.github/workflows/docker-branch.yml

```diff
@@ -0,0 +1,21 @@
+name: Build Docker image (non-master branches)
+on:
+  push:
+    branches:
+      - '!master'
```

Comment:
The trigger should be `pull-request` and not `push`. `tags: shaarli/shaarli:$GITHUB_REF` should also be adjusted (shuld be in the form `shaarli/shaarli:pr-PRNUMBER`)